### ISSUE TITLE
Add PPC64LE port

### ISSUE THREAD

**Issue by edelsohn** [LuaJIT/LuaJIT#42]
Depends on #25 

Sponsorship available.

**Comment by gut** [comment]
The Pull Request is a finalized port from our team. Please take a look.

**Comment by edelsohn** [comment]
Mike Pall commented that the port was not in a state ready for merge and release.  A sponsorship is available for a LuaJIT maintainer to bring the patches to a state that can be merged into LuaJIT development master.

**Comment by iHateInventNames** [comment]
@gut,
I advice you to try SET_END Linux function http://stackoverflow.com/questions/28725421/is-it-possible-to-change-endianness-mid-execution-on-arm-android-linux

Check for the SET_END availability in linux.h as mandatory for big endian target architecture while configure time.

This way you can set endiannes before Lua execution.

**Comment by gut** [comment]
But why would ppc64 change endianess to big? Are you focusing other systems? E.g: [Debian only supports little endian] (https://www.debian.org/releases/stable/i386/ch02s01.html.en)

**Comment by iHateInventNames** [comment]
Hi @gut,
I am not related neither to IBM nor LUA. Its just my advice. The patch set could not be merged because of not supported big endianness. This is not full support of PPC64. You can resolve it to earn the bounty https://www.bountysource.com/issues/25924774-add-ppc64le-port

**Comment by edelsohn** [comment]
@iHateInventNames Mike Pall's feedback had nothing to do with endianness nor about supporting PPC64 big endian.  His comments were about the quality of the patch.

http://www.freelists.org/post/luajit/PPC64le-port-status,1

**Comment by iHateInventNames** [comment]
Oh, I am sory for that. So BE seems could be an additional feature.
Nevermind.

On Sat, Oct 24, 2015 at 1:49 AM, David Edelsohn <notifications@github.com>
wrote:

> @iHateInventNames <https://github.com/iHateInventNames> Mike Pall's
> feedback had nothing to do with endianness nor about supporting PPC64 big
> endian. His comments were about the quality of the patch.
>
> http://www.freelists.org/post/luajit/PPC64le-port-status,1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LuaJIT/LuaJIT/issues/42#issuecomment-150710297>.
>



-- 

Sergei Krivonos
skype: sergio_krivonos

**Comment by pmatos** [comment]
I have been investigating the avenues for creating a testsuite for LuaJIT. I will be working on that and this issue in parallel soon so that we can in the end have a baseline to test the new backend against.

**Comment by leitao** [comment]
Any feadback about this issue? We are engaged to fix any issue that is required to have ppc64le accepted in Luajit.

**Comment by pmatos** [comment]
@leitao I have started looking into this late last year but after some research I don't feel qualified enough to solve this at the moment. I have don't have the time to go through the learning curve to get it done. Apologies.

**Comment by leitao** [comment]
@pmatos no worries. I am wondering if there is someone looking at it, or, if this issue is abandoned at all.

**Comment by edelsohn** [comment]
Integrating the proposed LJ_GC64 patch probably is the wrong approach.  A developer has initiated a pull request for a patch to start on the path to a !LJ_GC64 approach, but there has been no response.

**Comment by dmiller423** [comment]
So this is still open bounty and I won't be stepping on anyone toes?

**Comment by edelsohn** [comment]
The bounty is open in theory, but we are pursuing a different path while we wait for the LuaJIT community to respond to the existing pull requests.

**Comment by dmiller423** [comment]
Sorry I misunderstood or misread,  if it's likely to be accepted there is no point.

**Comment by edelsohn** [comment]
Are you interested to work on other bounties related to the POWER architecture or solely interested in LuaJIT?

**Comment by dmiller423** [comment]
Honestly,  I'm just out of work at the moment,  and capable.

**Comment by edelsohn** [comment]
Please contact me directly and we can discuss some of the other potential bounty projects.

**Comment by joshafest** [comment]
Is this issue still open?
I mean i saw this in bouty

**Comment by edelsohn** [comment]
The meta-issue of PPC64 support still is open, but the specific request to merge the earlier patch from IBM will not be pursued.  There has been progress to build the PPC32 port in 64 bit mode, similar to x86 built in x86-64 mode, with !LJ_GC64.  The work on the JIT seems to have stalled.

https://github.com/koriakin/LuaJIT

I am interested to see that completed and merged into the official LuaJIT sources.

**Comment by edelsohn** [comment]
I have adjusted the issue to completion of the JIT work. Marcin has completed the PPC64 LE Linux !LJ_GC64 interpreter.  This issue now is a request to complete enablement and implementation of the !LJ_GC64 JIT in some repository.  There is a separate issue and bounty for the implementation to be merged into the official LuaJIT repository.

**Comment by edelsohn** [comment]
@NukeRusich @DemiMarie Sponsorship is available for this issue. Are you interested in working on it?

**Comment by UplinkCoder** [comment]
@edelsohn I'd like to work on this.

**Comment by edelsohn** [comment]
@UplinkCoder Great. Looking forward to your progress!

**Comment by gut** [comment]
@UplinkCoder any news?

**Comment by joshafest** [comment]
any news about this issue? is it still open?

**Comment by edelsohn** [comment]
This is still open.  If someone wants to complete it, the bounty remains available.

**Comment by DavidUser** [comment]
That can be tested using this docker image https://hub.docker.com/r/ppc64le/ubuntu/ ?

**Comment by grooverdan** [comment]
@DavidUser, if you have ppc64le host kernel then sure. If you don't have Power8/9 hardware I suggest using qemu-system-ppc as an emulator. This will get to a functional userpace. To aid compile on non-emulated hardware there is https://developer.ibm.com/linuxonpower/advance-toolchain/. For cross compilation https://git.buildroot.org/buildroot/tree/package/luajit/luajit.mk might help to show what LuaJIT compile options are needed to cross compile.

**Comment by gut** [comment]
@DavidUser Or you may also use the minicloud system to grab a free power8 virtual machine:
http://openpower.ic.unicamp.br/minicloud/

**Comment by issacnitin** [comment]
Is the issue solved by #54 ?

**Comment by elliottslaughter** [comment]
If I'm looking to run LuaJIT on PPC64le today, is https://github.com/koriakin/LuaJIT still the right place to look? I see the two oustanding pull requests, but from the discussion here it sound like #54 isn't going to be merged, and #140 also hasn't had any feedback recently.

**Comment by edelsohn** [comment]
No one has been able to elicit a response from Mike and the LuaJIT community, so I have given up.

**Comment by runlevel5** [comment]
@MikePall I think many have attempted to do the port but I guess no one is more qualified than you. So if you are to take on the work, would you be able to give us a good idea how much funding for the bounty so the community can crowdfund?
